pkg/webhook/controlplane/ensurer.go

```diff
@@ -50,13 +47,11 @@ func NewEnsurer(logger logr.Logger) genericmutator.Ensurer {
 
 type ensurer struct {
 	genericmutator.NoopEnsurer
-	client client.Client
 	logger logr.Logger
 }
 
 // InjectClient injects the given client into the ensurer.
-func (e *ensurer) InjectClient(client client.Client) error {
-	e.client = client
+func (e *ensurer) InjectClient(_ client.Client) error {
```

Comment:
You could remove the whole `InjectClient` func as the ensurer no longer needs a client.